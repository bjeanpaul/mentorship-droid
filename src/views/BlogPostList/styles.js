import { StyleSheet } from 'react-native';

import colors from 'src/constants/colors';
import { FONT, FONT_WEIGHT } from 'src/constants/styles';


export default StyleSheet.create({
  blogPostList: {
    flex: 1,
    backgroundColor: colors.BG_DEFAULT_SECONDARY,
  },
  blogPost: {
    paddingVertical: 18,
    paddingHorizontal: 10,
    marginBottom: 8,
    flexDirection: 'row',
    backgroundColor: colors.BG_DEFAULT,
  },
  blogPostDetail: {
    flex: 1,
    flexDirection: 'column',
    alignItems: 'flex-start',
    paddingRight: 18,
  },
  blogPostCreatedAt: {
    fontSize: 11,
    color: colors.TEXT_SECONDARY,
    fontFamily: FONT.MEDIUM,
    fontWeight: FONT_WEIGHT.MEDIUM,
  },
  blogPostTitle: {
    paddingLeft: 6,
  },
  blogPostThumbnail: {
    flex: 1,
    flexDirection: 'column',
    alignItems: 'flex-end',
    paddingRight: 6,
  },
  blogPostThumbnailImage: {
    width: 96,
    height: 96,
    paddingRight: 6,
  },
});
