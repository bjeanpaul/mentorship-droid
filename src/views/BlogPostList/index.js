import moment from 'moment';
import React, { PropTypes } from 'react';
import { View, Image, TouchableNativeFeedback, ScrollView } from 'react-native';

import styles from './styles';
import { ImageUrl } from 'src/api/imageUrl';
import { BaseView, Header, Text } from 'src/components';


const BlogPostList = ({
  blogPosts,
  onBlogPostPress,
}) => (
  <BaseView>
      <Header style={Header.themes.dark}>
        <Text style={[Text.types.title, Text.themes.light]}>
          Blog
        </Text>
      </Header>

      <ScrollView>
        <View style={styles.blogPostList}>
          {blogPosts.map(blog => (
            <TouchableNativeFeedback
              key={blog.id}
              uid={`blogPosts:${blog.id}`}
              onPress={() => onBlogPostPress(blog.id)}
            >
              <View style={styles.blogPost}>
                <View style={styles.blogPostDetail}>
                  <Text style={styles.blogPostCreatedAt}>
                    {moment(blog.createdAt).fromNow().toUpperCase()}
                  </Text>

                  <Text style={[Text.types.secondaryTitle, styles.blogPostTitle]}>
                    {blog.title}
                  </Text>
                </View>

                <View style={styles.blogPostThumbnail}>
                  <Image
                    style={styles.blogPostThumbnailImage}
                    source={blog.thumbnail.resize(96, 96).toSource()}
                  />
                </View>
              </View>
            </TouchableNativeFeedback>
          ))}
        </View>
      </ScrollView>
  </BaseView>
);


BlogPostList.propTypes = {
  blogPosts: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    thumbnail: PropTypes.instanceOf(ImageUrl),
  })).isRequired,

  onBlogPostPress: PropTypes.func.isRequired,
};


export default BlogPostList;
