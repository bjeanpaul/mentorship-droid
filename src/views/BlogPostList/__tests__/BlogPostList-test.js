import React from 'react';

import BlogPostList from 'src/views/BlogPostList';


describe('BlogPostList', () => {
  function createComponent(props = {}) {
    return (
      <BlogPostList
        {...props}
      />
    );
  }

  it('should render', () => {
    expect(render(createComponent())).toMatchSnapshot();
  });
});
