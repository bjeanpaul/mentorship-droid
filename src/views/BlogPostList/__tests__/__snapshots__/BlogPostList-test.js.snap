exports[`BlogPostList should render 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
      },
      undefined,
    ]
  }>
  <View
    style={
      Array [
        Object {
          "height": 72,
          "justifyContent": "center",
          "paddingLeft": 16,
          "paddingRight": 16,
        },
        Object {
          "backgroundColor": "#003035",
        },
      ]
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#003035",
            "fontFamily": "Brandon Text",
          },
          Object {
            "fontSize": 13,
            "textAlign": "center",
          },
          Array [
            Object {
              "fontFamily": "Brandon Text_medium",
              "fontSize": 20,
              "fontWeight": "400",
            },
            Object {
              "color": "#fff",
            },
          ],
        ]
      }>
      Blog
    </Text>
  </View>
  <ScrollView>
    <View
      style={
        Object {
          "backgroundColor": "#eff2f2",
        }
      }>
      <TouchableNativeFeedback
        onPress={[Function]}
        uid="blogPosts:1">
        <View
          style={
            Object {
              "backgroundColor": "#fff",
              "flexDirection": "row",
              "marginBottom": 8,
              "paddingHorizontal": 10,
              "paddingVertical": 18,
            }
          }>
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flex": 1,
                "flexDirection": "column",
                "paddingRight": 18,
              }
            }>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#003035",
                    "fontFamily": "Brandon Text",
                  },
                  Object {
                    "fontSize": 13,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "#9fb1b3",
                    "fontFamily": "Brandon Text_medium",
                    "fontSize": 11,
                    "fontWeight": "400",
                  },
                ]
              }>
              IN 2 MONTHS
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#003035",
                    "fontFamily": "Brandon Text",
                  },
                  Object {
                    "fontSize": 13,
                    "textAlign": "center",
                  },
                  Array [
                    Object {
                      "fontFamily": "Brandon Text_medium",
                      "fontSize": 18,
                      "fontWeight": "400",
                      "textAlign": "left",
                    },
                    Object {
                      "paddingLeft": 6,
                    },
                  ],
                ]
              }>
              Foo
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "flex": 1,
                "flexDirection": "column",
                "paddingRight": 6,
              }
            }>
            <Image
              source={
                Object {
                  "uri": "/api/foo.png?filter_spec=min-192x192",
                }
              }
              style={
                Object {
                  "height": 96,
                  "paddingRight": 6,
                  "width": 96,
                }
              } />
          </View>
        </View>
      </TouchableNativeFeedback>
      <TouchableNativeFeedback
        onPress={[Function]}
        uid="blogPosts:2">
        <View
          style={
            Object {
              "backgroundColor": "#fff",
              "flexDirection": "row",
              "marginBottom": 8,
              "paddingHorizontal": 10,
              "paddingVertical": 18,
            }
          }>
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flex": 1,
                "flexDirection": "column",
                "paddingRight": 18,
              }
            }>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#003035",
                    "fontFamily": "Brandon Text",
                  },
                  Object {
                    "fontSize": 13,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "#9fb1b3",
                    "fontFamily": "Brandon Text_medium",
                    "fontSize": 11,
                    "fontWeight": "400",
                  },
                ]
              }>
              IN 2 MONTHS
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#003035",
                    "fontFamily": "Brandon Text",
                  },
                  Object {
                    "fontSize": 13,
                    "textAlign": "center",
                  },
                  Array [
                    Object {
                      "fontFamily": "Brandon Text_medium",
                      "fontSize": 18,
                      "fontWeight": "400",
                      "textAlign": "left",
                    },
                    Object {
                      "paddingLeft": 6,
                    },
                  ],
                ]
              }>
              Bar
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "flex": 1,
                "flexDirection": "column",
                "paddingRight": 6,
              }
            }>
            <Image
              source={
                Object {
                  "uri": "/api/bar.png?filter_spec=min-192x192",
                }
              }
              style={
                Object {
                  "height": 96,
                  "paddingRight": 6,
                  "width": 96,
                }
              } />
          </View>
        </View>
      </TouchableNativeFeedback>
    </View>
  </ScrollView>
</View>
`;
